src/components/YourCompany/NameTranslations/NameTranslation.vue

```diff
@@ -0,0 +1,360 @@
+                @click="cancelNameTranslationCorrection()"
+                :disabled="isAddingNameTranslation"
+              >
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-flex>
+        </v-layout>
+      </v-flex>
+    </v-layout>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+
+// Components
+import { ConfirmDialog } from '@/components/dialogs'
+import { ListNameTranslation, AddNameTranslation } from '.'
+
+// Interfaces
+import { ActionBindingIF, ConfirmDialogType,
+  NameTranslationDraftIF, NameTranslationIF } from '@/interfaces'
+
+// Enums
+import { ActionTypes } from '@/enums'
+
+// Mixins
+import { CommonMixin } from '@/mixins'
+import { Action, Getter } from 'vuex-class'
+
+@Component({
+  components: {
+    AddNameTranslation,
+    ListNameTranslation,
+    ConfirmDialog
+  }
+})
+export default class NameTranslation extends Mixins(CommonMixin) {
+  // Refs
+  $refs!: {
+    confirmTranslationDialog: ConfirmDialogType
+  }
+
+  @Prop({ default: () => [] })
+  private nameTranslations!: NameTranslationDraftIF[]
+
+  private actionTypes = ActionTypes
+
+  // Properties
+  private draftTranslations: NameTranslationDraftIF[] = []
+  private isEditing: boolean = false
+  private isAddingNameTranslation = false
+  private editingNameTranslation = ''
+  private editIndex = -1
+
+  private get hasPendingChange (): boolean {
+    return this.draftTranslations.length !== this.nameTranslations.length ||
+      !this.draftTranslations.every((translation, index) => {
+        return this.nameTranslations[index].value === translation.value
+      })
+  }
+
+  private get hasNameTranslationChange (): boolean {
+    return this.draftTranslations.length > 0 &&
+      this.draftTranslations.filter(x => x.action).length > 0
+  }
+
+  private setNameTranslations (): void {
+    this.emitNameTranslations(this.draftTranslations)
+    this.emitHaveChanges(this.hasNameTranslationChange)
+    this.isEditing = false
+  }
+
+  private resetNameTranslations (): void {
+    this.draftTranslations = this.nameTranslations
+      .filter(x => x.action !== 'added')
+      .map(a => {
+        const translation = Object.assign({}, a)
+        translation.value = translation.oldValue || translation.value
+        translation.oldValue = null
+        translation.action = null
+        return translation
+      })
+    this.emitNameTranslations(this.draftTranslations)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+
+  private cancelNameTranslationCorrection () {
+    const nameTranslations = this.nameTranslations || []
+    const hasUnsavedData = this.draftTranslations.length !== nameTranslations.length ||
+      !this.draftTranslations.every((translation, index) => {
+        return nameTranslations[index].value === translation.value &&
+          nameTranslations[index].action === translation.action
```

Comment:
I had to look at this code for a couple of minutes to understand it, haha.
Could you please add a comment saying what you're looking for to determine if there is unsaved data?

Reply:
Great comment, thanks!